transaction/transactor.go

```diff
@@ -127,9 +130,18 @@ func Log(logger *zap.Logger, reducedLoggingResponseCodes []int) kithttp.ServerFi
 		tid, _ := ctx.Value(ContextKeyRequestTID).(string)
 
 		requestArrival, ok := ctx.Value(ContextKeyRequestArrivalTime).(time.Time)
+		logger := sallust.Get(ctx)
 
+		logger = logger.With(
+			zap.Any("deviceid", GetDeviceId(r)),
+		)
+
+		// ctx = sallust.With(ctx, logger)
 		if !ok {
-			logger = logger.With(zap.Any("duration", time.Since(requestArrival)))
+			logger = logger.With(
+				zap.Any("duration", time.Since(requestArrival)),
+				zap.Any("deviceid", GetDeviceId(r)),
```

Comment:
Is this needed since it's already there from line 136?

Reply:
had a review with owen and serena and this is not the correct solution. so this line will be deleted once i come up with the correct solution.